### DISCUSSION TITLE
NativeAOT with Mono exports?

### DISCUSSION THREAD

**Discussion by slxdy** [dotnet/runtime#94148] [Q&A]
Ok, this may sound stupid, but hear me out.
Would it be possible to have the Mono API in my NativeAOT-generated module? I'd like to do something similar to Unity's IL2CPP, but I'm not sure if this is possible.
My final idea is to replace Unity's IL2CPP with NativeAOT, which would require the Mono exports that UnityPlayer interacts with.

**Comment by vitek-karas** [comment]
NativeAOT is a different implementation of the runtime, so providing the same functionality mono does via its native exports would be pretty difficult. Especially since mono's native exports tend to leak the runtime implementation details - all of that would have to be emulated, likely loosing lot of performance.

**Comment by jkotas** [comment]
Unity team has been working towards enabling CoreCLR and NativeAOT with their engine for a while. You can check the progress at https://github.com/Unity-Technologies/runtime/

**Comment by slxdy** [comment]
Oh man, another unity .net runtime fork. Lord have mercy.

**Comment by slxdy** [comment]
Anyway, thanks for all the responses. I think I will give up and wait for unity to do this themselves.
